src/posix.c

```diff
@@ -216,9 +217,48 @@ int p_write(git_file fd, const void *buf, size_t cnt)
 	return 0;
 }
 
-#ifdef NO_MMAP
+ssize_t p_pread(git_file fd, void *buf, size_t nbyte, git_off_t offset)
+{
+#ifdef GIT_WIN32
+	/* TODO: Implement using Win32 overlapped IO to avoid updating fd */
 
-#include "map.h"
+	ssize_t error;
+	const git_off_t cpos = p_lseek(fd, 0, SEEK_CUR);
+
+	if ((error = p_lseek(fd, offset, SEEK_SET) < 0 ? -1 : 0))
+		goto cleanup;
+
+	if ((error = p_read(fd, buf, nbyte)) < 0)
```

Comment:
We probably want out function to return `-1` on any failure, not propagate `p_read`'s failure value

Reply:
Since this is win32 specific code, I moved it to `win32/posix_w32.c` and used `mmap` based IO for emulating `p_pread` and `p_pwrite`